pkg/ring/kv/etcd/etcd.go

```diff
@@ -16,9 +19,14 @@ import (
 
 // Config for a new etcd.Client.
 type Config struct {
-	Endpoints   []string      `yaml:"endpoints"`
-	DialTimeout time.Duration `yaml:"dial_timeout"`
-	MaxRetries  int           `yaml:"max_retries"`
+	Endpoints          []string      `yaml:"endpoints"`
+	DialTimeout        time.Duration `yaml:"dial_timeout"`
+	MaxRetries         int           `yaml:"max_retries"`
+	EnableTLS          bool          `yaml:"tls_enabled"`
+	CertFile           string        `yaml:"tls_cert_file"`
+	KeyFile            string        `yaml:"tls_key_file"`
+	TrustedCAFile      string        `yaml:"tls_client_ca_file"`
```

Comment:
Sorry for being annoying but I would like to keep consistency with existing TLS config (eg. see https://cortexmetrics.io/docs/configuration/configuration-file/#ruler_config).

```suggestion
	CertFile           string        `yaml:"tls_cert_path"`
	KeyFile            string        `yaml:"tls_key_path"`
	TrustedCAFile      string        `yaml:"tls_ca_path"`
```